### PR TITLE
Refactored and improved seeds

### PR DESCRIPTION
**Pull request by lucasbordeau**
- Added a new Seeder service to help with custom object seeds
- Added RichTextFieldInput to edit a rich text field directly on the table, but deactivated it for now.

### CODE REVIEW

File: packages/twenty-front/src/modules/object-record/record-field/types/FieldMetadata.ts [hunk 1]
```diff
@@ -46,95 +44,85 @@ export type FieldDateMetadata = {
...
-    type?: 'percentage' | 'number';
+    type?: FieldNumberVariant;
   };
 };
 
 export type FieldLinkMetadata = {
   objectMetadataNameSingular?: string;
   placeHolder: string;
   fieldName: string;
-  settings?: Record<string, never>;
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/object-record/record-field/types/FieldMetadata.ts:64]: already done in another PR, let's see who merges first

File: packages/twenty-server/src/engine/workspace-manager/demo-objects-prefill-data/seed-workspace-member-with-demo-data.ts [hunk 2]
```diff
@@ -8,7 +8,7 @@ export const DEMO_SEED_WORKSPACE_MEMBER_IDS = {
   TIM: '20202020-1553-45c6-a028-5a9064cce07e',
 };
 
-export const workspaceMemberPrefillData = async (
+export const seedWorkspaceMemberWithDemoData = async (
```
Comment (charlesBochet) [inline packages/twenty-server/src/engine/workspace-manager/demo-objects-prefill-data/seed-workspace-member-with-demo-data.ts:11]: we move it out from prefill data, aren't we breaking workspace creation?

File: packages/twenty-server/src/modules/company/standard-objects/company.workspace-entity.ts [hunk 3]
```diff
@@ -48,6 +51,20 @@ export const SEARCH_FIELDS_FOR_COMPANY: FieldTypeAndNameMetadata[] = [
   { name: DOMAIN_NAME_FIELD_NAME, type: FieldMetadataType.LINKS },
 ];
 
+export enum CompanyDemoSelect {
```
Comment (charlesBochet) [inline packages/twenty-server/src/modules/company/standard-objects/company.workspace-entity.ts:54]: this should not go into the company.workspace-entity.ts

**Comment by FelixMalfait** [comment]
Small comment from @Bonapara: the default workspace of Tim Apple should be Apple and not ACME

**Comment by Bonapara** [comment]
Maybe we could use a fake logo to make it more fun:

image

File: packages/twenty-server/src/engine/workspace-manager/demo-objects-prefill-data/seed-workspace-member-with-demo-data.ts (hunk 2 above)
Comment (lucasbordeau) [reply packages/twenty-server/src/engine/workspace-manager/demo-objects-prefill-data/seed-workspace-member-with-demo-data.ts:11]: I need to check

File: packages/twenty-server/src/engine/metadata-modules/relation-metadata/hooks/before-delete-one-relation.hook.ts [hunk 4]
```diff
@@ -25,12 +30,12 @@ export class BeforeDeleteOneRelation implements BeforeDeleteOneHook {
...
-      await this.relationMetadataService.findOneWithinWorkspace(workspaceId, {
-        where: {
-          id: instance.id.toString(),
-        },
-      });
+    const relationMetadata = await this.relationMetadataRepository.findOne({
+      where: {
+        workspaceId,
+        id: instance.id.toString(),
+      },
```
Comment (Weiko) [inline packages/twenty-server/src/engine/metadata-modules/relation-metadata/hooks/before-delete-one-relation.hook.ts:37]: If you want to use repository, keep in mind that relations won't be loaded by default
```suggestion
      },
      relations: ['fromFieldMetadata', 'toFieldMetadata'],
```
otherwise it will fail below when retrieving the fields in relationMetadata.toFieldMetadata.isCustom for example

File: packages/twenty-server/src/engine/seeder/data-seeds/pets-data-seeds.ts [hunk 5]
```diff
@@ -0,0 +1,2852 @@
+export const PETS_DATA_SEEDS = [
```
Comment (Weiko) [inline packages/twenty-server/src/engine/seeder/data-seeds/pets-data-seeds.ts:1]: I think we should add large files like this in ignorePatterns of our eslint config. Check eslintrc.cjs we should already have some

File: packages/twenty-server/src/engine/seeder/seeder.service.ts [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+      objectMetadataAfterFieldCreation.fields.some(
+        (f) => f.name === field.name || f.name === `name`,
+      ),
+    );
+
+    if (filteredFields.length === 0) {
+      throw new Error('No fields found for seeding, check metadata');
+    }
+
+    filteredFields.unshift({
```
Comment (Weiko) [inline packages/twenty-server/src/engine/seeder/seeder.service.ts:77]: I think "name" should already be there by the createOneObject?

File: packages/twenty-server/src/engine/seeder/seeder.service.ts [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+      name: 'name',
+      type: FieldMetadataType.TEXT,
+      label: 'Name',
+    });
+
+    const fieldMetadataMap = filteredFields
+      .map((field) => {
```
Comment (Weiko) [inline packages/twenty-server/src/engine/seeder/seeder.service.ts:84]: nit: You should be able to use flatMap to avoid map().flat()
```suggestion
      filteredFields
      .flatMap((field) => {
        if (!isCompositeFieldMetadataType(field.type)) {
          return field.name;
        }

        const compositeFieldTypeDefinition = compositeTypeDefinitions.get(
          field.type,
        );

        if (!isDefined(compositeFieldTypeDefinition)) {
          throw new Error(
            `Composite field type definition not found for ${field.type}`,
          );
        }

        return (
          compositeFieldTypeDefinition.properties?.map(
            (property) => `${field.name}${capitalize(property.name)}`,
          ) ?? []
        );
      })
      .filter(isDefined);
```

File: packages/twenty-server/src/engine/seeder/seeder.service.ts [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+        }
+      })
+      .flat()
+      .filter(isDefined);
+
+    const flattenedSeeds = dataSeeds.map((seed) => {
+      const flattenedSeed = {};
+
+      for (const field of filteredFields) {
+        if (isCompositeFieldMetadataType(field.type)) {
```
Comment (Weiko) [inline packages/twenty-server/src/engine/seeder/seeder.service.ts:117]: nit: Could be refactored, looks like we are doing the same checks here and above

File: packages/twenty-server/src/engine/seeder/seeder.service.ts [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+
+@Injectable()
+export class SeederService {
+  constructor(
+    private readonly objectMetadataService: ObjectMetadataService,
+    private readonly fieldMetadataService: FieldMetadataService,
+    private readonly workspaceDataSourceService: WorkspaceDataSourceService,
+  ) {}
+
+  public async seedCustomObjects(
```
Comment (Weiko) [inline packages/twenty-server/src/engine/seeder/seeder.service.ts:22]: nit: This method seems to have a lot of responsibilities, could be refactored in smaller functions maybe?

File: packages/twenty-server/src/engine/workspace-manager/demo-objects-prefill-data/seed-workspace-member-with-demo-data.ts (hunk 2 above)
Comment (Weiko) [reply packages/twenty-server/src/engine/workspace-manager/demo-objects-prefill-data/seed-workspace-member-with-demo-data.ts:11]: Any update on this?

File: packages/twenty-server/src/modules/workflow/common/standard-objects/workflow.workspace-entity.ts [hunk 10]
```diff
@@ -49,7 +50,7 @@ const WorkflowStatusOptions = [
     value: WorkflowStatus.DEACTIVATED,
     label: 'Deactivated',
     position: 2,
-    color: 'grey',
+    color: 'gray',
```
[the remark this answers is not in the document]
Comment (lucasbordeau) [reply packages/twenty-server/src/modules/workflow/common/standard-objects/workflow.workspace-entity.ts:53]: There's a strong reason yes, it breaks some parts of the code as grey is not recognized in the front end and the tags end up with no color.

File: packages/twenty-server/src/engine/seeder/data-seeds/pets-data-seeds.ts (hunk 5 above)
Comment (lucasbordeau) [reply packages/twenty-server/src/engine/seeder/data-seeds/pets-data-seeds.ts:1]: Ok

File: packages/twenty-front/src/modules/object-record/record-field/components/FieldInput.tsx [hunk 11]
```diff
@@ -167,8 +165,6 @@ export const FieldInput = ({
           onTab={onTab}
           onShiftTab={onShiftTab}
         />
-      ) : isFieldRichText(fieldDefinition) ? (
```
[the remark this answers is not in the document]
Comment (lucasbordeau) [reply packages/twenty-front/src/modules/object-record/record-field/components/FieldInput.tsx:170]: Yes we want to desactivate this field input for now, and activate it later in another PR

File: packages/twenty-server/src/engine/workspace-manager/demo-objects-prefill-data/seed-workspace-member-with-demo-data.ts (hunk 2 above)
Comment (lucasbordeau) [reply packages/twenty-server/src/engine/workspace-manager/demo-objects-prefill-data/seed-workspace-member-with-demo-data.ts:11]: It's ok

**Review by Weiko — APPROVED** [review]
